### DISCUSSION TITLE
Blacklisting users or subreddits

### DISCUSSION THREAD

**Discussion by thany** [MalloyDelacroix/DownloaderForReddit#268] [Q&A] [answered]
When downloading subreddits, I would like to be able to blacklist a set of users that might have posted in those subreddits. And vice versa: when downloading users, you might want to be able to blacklist a set of subreddits where those users may have posted.

Seems like a pretty simple thing to have around. So simple in fact, that I believe I must be missing something. Blocking posts from certain users to get downloaded seems *so* basic in fact, and *so* blatantly obvious in certain communities, that I can't believe this program doesn't have a facility to do that. I *must* be missing something, right?

If the program doesn't already have this feature, then it's time to add it 😀
Seems like a basic multiline text field should do just fine.

If it does already have this feature, then where?

**Comment by MalloyDelacroix** [comment, marked as answer]
It's true, it doesn't currently have this feature.  I have to agree, that was a major oversight when developing version 3.  It's one of the first features I plan to add when I have time to do some major work on the app.

One thing that's been holding me back is that I'm not sure where to put the blacklist.

One option is to be able to make any of the major lists a blacklist, then add an checkbox under the download button to "exclude downloads from user list" or "exclude users from subreddit list".  This will complicate the main UI and no doubt lead to come confusion about which list is selected.

Another option is to include a per user/subreddit blacklist available in the user/subreddit settings.  Seems that a blacklist that sticks with a user/subreddit would be beneficial.  But, this will clutter an already intense settings dialog and will not help the confusion already present.

Both of these options have problems and impacts on the UI.  It has been brought to my attention more and more lately that the UI is already confusing, and I'm afraid these options will only make it worse.  But I do agree, this feature is needed.

I am certainly open to suggestions on the best way to integrate a blacklist.

**Reply by thany** [reply]
The whole UI needs a big overhaul anyway. It's become far too confusing with v3, to the point that I have no idea what I'm doing at certain points. If you want help, I'm not a UX designer, but I have gotten some affinity for it while working with UX designer colleagues. So, lemme know?

**Comment by zacker150** [comment]
I think a better way would be to add a whitelists and blacklists field to each list. These fields would contain a list of one-way references to other lists. 

Example: 

List A:
* Items: 
  * /u/user1
  * /u/user2
* Blacklists:
  * List B 

List B: 
* Items: 
  * /r/EarthPorn
  * /r/pics
  * /r/funny
* Blacklists: 
  * List A

When you download List A, it downloads all posts made by user1 and user2 except the ones in /r/EarthPorn, /r/pics, and /r/funny. 

When you download List B, it downloads all posts in /r/EarthPorn, /r/pics, and /r/funny except those made by user1 and user2.

**Reply by MalloyDelacroix** [reply]
So a hybrid approach where we would use main lists as a blacklist, but the blacklist selection would would be made on a per download list basis, instead of a per download session basis.  I like this idea.

Maybe the UI is something like this:

![Blacklist Example](https://user-images.githubusercontent.com/22173536/151710547-1c80e899-1f41-45a4-a5b2-61f38a75e872.jpg)
